satellite/peer.go

```diff
@@ -509,6 +517,22 @@ func New(log *zap.Logger, full *identity.FullIdentity, db DB, config *Config, ve
 		)
 	}
 
+	{ // setup referral offers
+		log.Debug("Setting up offers")
+		referralConfig := config.Referral
+
+		peer.Referral.Listener, err = net.Listen("tcp", referralConfig.Address)
+		if err != nil {
+			return nil, errs.Combine(err, peer.Close())
+		}
+
+		peer.Referral.Endpoint = offersweb.NewServer(
```

Comment:
Wondering whether we should have `admin.Server` instead of offersweb... a more general purpose admin interface. I feel like we might get more of these admin panels.

The question then is, how will the permissions be managed. Or whether it makes more sense to combine this with the `console` itself?

Reply:
Note, I think this might be fine solution for now, but I'm not sure whether there's a nicer solution that doesn't take much longer, but saves us time in the future.

Reply:
I added a middleware for checking request address

Reply:
I checked in with @jtolds and @brimstone. I think for now we will rely on the devops to restrict access to this port through VPN. In that case, the requests won't be coming in from localhost